### PR TITLE
feat(wizard): added XR setup wizard TRNG-1057

### PR DESCRIPTION
**Pull request by Gusinuhe**
### Description

**Fixes**: Implements the XR setup wizard.

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Editor/PackageManager/PackageDependencies/XRProviders/OculusXRPackageEnabler.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            OnPackageEnabled += InitializeOculusLoader;
+        }
+
+        public void Dispose()
+        {
+            OnPackageEnabled -= InitializeOculusLoader;
+        }
+
+        private void InitializeOculusLoader(object sender, EventArgs e)
```
Comment (SimonTheSourcerer) [inline Editor/PackageManager/PackageDependencies/XRProviders/OculusXRPackageEnabler.cs:30]: code duplication, can you created a base which is handling all this reflection stuff?

**Comment by Gusinuhe** [comment]
> When importing the creator (core) in a clean project the following error pops up and therefore the Wizard cannot be opened:
> 
> `Assets\Core\Editor\PackageManager\PackageDependencies\XRProviders\XRProvider.cs(3,22): error CS0234: The type or namespace name 'Management' does not exist in the namespace 'UnityEngine.XR' (are you missing an assembly reference?)`
> 
> After restarting Unity the package is automatically added. Did you encounter the same issue?

Fixed

**Comment by tomwim** [comment]
Is this ready for review or more going on?

**Comment by Gusinuhe** [comment]
> Is this ready for review or more going on?

Due to https://github.com/ValveSoftware/unity-xr-plugin/issues/16 I will temporary disable Ooenvr XR and use the legacy one, at least for Unity 2019.4.

**Comment by Gusinuhe** [comment]
Ready to review 👍 

I added support for the legacy OpenVR so VRTK would still work in this version.
Unity 2020 will use the OpenVR XR plugin, hoping that https://github.com/ValveSoftware/unity-xr-plugin adds support for inputs.

**Comment by Maxsiom** [comment]
If you have any XR Plug-In installed (in my case Windows MR) and then use the wizard and choose HTC, legacy VR cannot be activated because the XR Plug-In is still active and installed. Can we deinstall and disable XR Management it beforehand?

**Comment by Gusinuhe** [comment]
> If you have any XR Plug-In installed (in my case Windows MR) and then use the wizard and choose HTC, legacy VR cannot be activated because the XR Plug-In is still active and installed. Can we deinstall and disable XR Management it beforehand?

You are right, the XR Management Plugin prevents the legacy XR configuration to be installed. 

In order to uninstall XR Management remove it from the Package Manager.

**Comment by Maxsiom** [comment]
Yeah, can this be done automatically via script or do you have any concerns?

**Comment by Gusinuhe** [comment]
> Yeah, can this be done automatically via script or do you have any concerns?

It can be done but seems quite some effort for such an edge case.
Our Dependency Manager does not handle removing packages.

**Comment by Maxsiom** [comment]
Yes, true. But can we gray out the option then and give any warning/info?

**Comment by Gusinuhe** [comment]
> Yes, true. But can we gray out the option then and give any warning/info?

It is more complex than it seems.
The XR management package, if installed, blocks the legacy API, and there is no way to use XR Management package API without having it installed, but again, that forces to not being able to install legacy OpenVR so we just assumed that none are installed.

The only non-conflictive way would be to checked the packages installed in the package manager, which is an asynchronous API that forces us to use EditorUpdate and currently we don´t have an implementation that allows us to do so.

I am not against doing it, we would need it at some point but that is another feature and could explode the effort for this one.
My suggestion is to do it in a follow-up issue.

**Comment by Gusinuhe** [comment]
Follow up ticket [TRNG-1078](https://jira.innoactive.de/browse/TRNG-1078)

**Pull request merged**